projects/python-contract-examples/smart_contracts/arc4_types/contract.py

```diff
@@ -0,0 +1,305 @@
+# pyright: reportMissingModuleSource=false
+import typing as t
+
+from algopy import (
+    ARC4Contract,
+    GlobalState,
+    String,
+    UInt64,
+    arc4,
+    urange,
+)
+from algopy.arc4 import abimethod
+
+
+class Arc4Types(ARC4Contract):
+
+    # example: ARC4_UINT64
+    @abimethod()
+    def add_arc4_uint64(self, a: arc4.UInt64, b: arc4.UInt64) -> arc4.UInt64:
+        """
+        Math operations (like a + b) are not supported on arc4.UInt64 types
+        since they are internally represented as byte arrays in the AVM.
+        Use the .native property to perform arithmetic operations.
+        """
+
+        # Use the native integers to perform arithmetic
+        c = a.native + b.native
+
+        return arc4.UInt64(c)
```

Comment:
For clarity, could add a brief comment that you are converting the type back to arc4.